rdr_service/gcloud_functions/genomic_aw1_jh_function/main.py

```diff
@@ -0,0 +1,94 @@
+#
+# This file is subject to the terms and conditions defined in the
+# file 'LICENSE', which is part of this source code package.
+#
+import logging
+
+from aou_cloud.services.gcp_cloud_function import GCPCloudFunctionContext, \
+    FunctionStoragePubSubHandler
+from aou_cloud.services.gcp_cloud_tasks import GCPCloudTask
+
+
+# Function name must contain only lower case Latin letters, digits or underscore. It must
+# start with letter, must not end with a hyphen, and must be at most 63 characters long.
+# There must be a python function in this file with the same name as the entry point.
+function_name = 'genomic_aw1_jh_function'
+
+# [--trigger-bucket=TRIGGER_BUCKET | --trigger-http | --trigger-topic=TRIGGER_TOPIC |
+# --trigger-event=EVENT_TYPE --trigger-resource=RESOURCE]
+# NOTE: Default function timeout limit is 60s, maximum can be 540s.
+deploy_args = [
+    '--trigger-resource=%%CLOUD_RESOURCE%%',
+    '--trigger-event google.storage.object.finalize',
+    '--timeout=540',
+    '--memory=512'
+]
+
+task_queue = 'resource-tasks'
+_logger = logging.getLogger('function')
+
+
+class GenomicManifestGenericFunction(FunctionStoragePubSubHandler):
+
+    def created(self):
+        """ Handle storage object created event. """
+        # Verify this is a file that we want to process.
+        if '_sample_manifests' not in self.event.name.lower():
+            _logger.info(f'Skipping file {self.event.name}, name does not match AW1 file.')
+            return
+
+        _logger.info(f"file found: {self.event.name}")
+
+        cloud_file_path = f'{self.event.bucket}/{self.event.name}'
+
+        data = {
+            "file_path": cloud_file_path,
+            "bucket_name": self.event.bucket,
+            "upload_date": self.event.timeCreated,
+        }
+
+        _logger.info("Pushing cloud task...")
+
+        _task = GCPCloudTask()
+        _task.execute('/resource/task/IngestAW1ManifestTaskApi', payload=data, queue=task_queue)
+
+
+def get_deploy_args(gcp_env):
+    """
+    Return the trigger and any additional arguments for the 'gcloud functions deploy' command.
+    Warning: function get_deploy_args() must come after all class definitions.
+    """
+    _project_suffix = gcp_env.project.split('-')[-1]
+
+    # Change these to appropriate buckets in derived functions for GCs' buckets
+    cloud_resource = 'aou-rdr-sandbox-mock-data'
+
+    if _project_suffix == 'sandbox':
+        cloud_resource = 'aou-rdr-sandbox-mock-data'
+
+    if _project_suffix == 'stable':
+        cloud_resource = 'genomics-baylor-dryrun'
+
+    if _project_suffix == 'prod':
+        cloud_resource = 'prod-genomics-baylor'
```

Comment:
I see that this is somewhat confusing. The buckets are named "baylor", but Baylor actually manages 2 genomic center IDs `gc_site_id`: JH and BCM. The AW1 files typically come through as JH (Johns Hopkins), and the AW2 files are returned as BCM (Baylor College of Medicine).  I tried to be consistent with my function naming by naming them based on the `gc_site_id`. The function name doesn't really matter, as long as they're unique, but it may be better to name them with the name of the bucket they are monitoring: i.e. broad, baylor, and northwest.